### PR TITLE
Add forecast for a specific day, including today

### PR DESCRIPTION
**Pull request by corus87**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,8 +16,10 @@ kalliope install --git-url https://github.com/kalliope-project/kalliope_neuron_o
 | api_key   | YES      | None    |                             | User API key of the OWM API                                                                       |
 | location  | YES      | None    |                             | The location                                                                                      |
 | lang      | No       | en      | multiple                    | First 2 letters cf : section Multilingual support in : [lang](https://openweathermap.org/current) |
-| temp_unit | No       | Kelvin  | Celsius, Kelvin, Fahrenheit |                                                                                                   |
-| country   | No       | US      | multiple                    |  Frist 2 letters of the country cf API doc                                                        |
```
Comment (Sispheor) [inline README.md]: add a link the the API doc would be useful then.

File: README.md [hunk 2]
```diff
@@ -16,8 +16,10 @@ kalliope install --git-url https://github.com/kalliope-project/kalliope_neuron_o
...
+| temp_unit | No       | Kelvin  | Celsius, Kelvin, Fahrenheit |                                                            |                                                      
+| country   | No       | US      | multiple                    | Frist 2 letters of the country cf API doc                  |
```
Comment (Sispheor) [inline README.md]: add a link the the API doc would be useful then.
Never seen that it was not present.

File: README.md [hunk 3]
```diff
@@ -84,6 +100,77 @@ Load the location from your order
...
+      - order: "wie wird das wetter am {{ day }}"
+    neurons:
+      - openweathermap:
+          api_key: "fdfba4097c318aed7836b2a85a6a05ef"
+          lang: "de"
+          country: "DE"
+          location: "{{ location }}"
+          day: "{{ day }}"
+          days:
+            Monday: "Montag"
```
Comment (Sispheor) [inline README.md]: case sensitive? If yes, need to be documented

File: README.md [hunk 4]
```diff
@@ -84,6 +100,77 @@ Load the location from your order
...
+          say_template: "Das wetter am {{ forecast_day }} wird {{ forecast_weather_descripton }} bei Temperaturen von {{ forecast_min_temp }} bis {{ forecast_max_temp }} Grad"
+```
+Forecast example for today
+```
+  - name: "weather-forecast-today"
+    signals:
+      - order: "How will be the weather today in {{ location }}"
+    neurons:
+      - openweathermap:
+          api_key: "fdfba4097c318aed7836b2a85a6a05ef"
```
Comment (Sispheor) [inline README.md:168]: hope this is not your real key

File: README.md [hunk 5]
```diff
@@ -16,8 +16,10 @@ kalliope install --git-url https://github.com/kalliope-project/kalliope_neuron_o
...
 | lang      | No       | en      | multiple                    | First 2 letters cf : section Multilingual support in : [lang](https://openweathermap.org/current) |
-| temp_unit | No       | Kelvin  | Celsius, Kelvin, Fahrenheit |                                                                                                   |
-| country   | No       | US      | multiple                    |  Frist 2 letters of the country cf API doc                                                        |
+| temp_unit | No       | Kelvin  | Celsius, Kelvin, Fahrenheit |                                                            |                                                      
+| country   | No       | US      | multiple                    | Frist 2 letters of the country cf API doc                  |
+| day       | No       | None    | today or a weekday          | To get the forecast for a specific day of the week         |
+| days      | No       | All weekdays in english |              | To ask the day in your language you need to translate them, look at the example |
```
Comment (Sispheor) [inline README.md]: should be renamed "days_translation" as it seems to be the idea. Explain that is mandatory if you use "day" value in another language than english. Also, you should deals with case sensitivity. i recommand to use lower() method on every entry the user can provide so you are sure.

File: openweathermap.py [hunk 6]
```diff
@@ -13,109 +16,160 @@ def __init__(self, **kwargs):
...
+        response = requests.get(url)
+        data = response.json()
+        
+        # To get the forecast for today we dont need self.days, so we update the dictionary with the current day
+        if self.day == 'today':
+            self.days.update({datetime.now().strftime('%A') : 'today'})
+
+        forecast = {}
+        forecast.update({'forecast_city' : data['city']['name']})
+        for f in data['list']:
```
Comment (Sispheor) [inline openweathermap.py]: don't use only a letter for your variables. Use comple words so we do know with what we are playing when reading the code.

File: openweathermap.py [hunk 7]
```diff
@@ -13,109 +16,160 @@ def __init__(self, **kwargs):
...
-                "clouds_coverage_tomorrow": clouds_coverage_tomorrow
-            }
-
-            self.say(message)
-
+                self.extended_location = self.location + "," + self.country
+            if not self.day:
+                owm = pyowm.OWM(API_key=self.api_key, language=self.lang)
+
+                # Tomorrow
```
Comment (Sispheor) [inline openweathermap.py]: is this part still needed then? Is still working?

File: README.md (hunk 2 above)
Comment (corus87) [reply README.md]: Done

File: openweathermap.py [hunk 8]
```diff
@@ -13,109 +16,160 @@ def __init__(self, **kwargs):
...
+                    "snow_tomorrow": snow_tomorrow,
+                    "rain_tomorrow": rain_tomorrow,
+                    "clouds_coverage_tomorrow": clouds_coverage_tomorrow
+                }
+
+                self.say(message)
+            # Daily forecast
+            if self.day:
+                forecast = self.GetForecast(extended_location)
+                self.say(forecast)
```
Comment (Sispheor) [inline openweathermap.py]: I don't really like the idea of return two different version of dict variables. I think it would be better to refactor the neuron to have a unique output.
If the "toomorow" feature is included in your update, then we can remove it from the first part.

File: README.md (hunk 3 above)
Comment (corus87) [reply README.md]: Its a dict, and it will be iterate through it with .lower()

File: README.md (hunk 4 above)
Comment (corus87) [reply README.md:168]: No, but maybe yours? I copied it from the orginal example ^^

File: README.md (hunk 5 above)
Comment (corus87) [reply README.md]: I renamed it, should be more clear now. And it iterate through the weekdays in english which are provided in the dict and I also use .lower() method.

**Comment by Sispheor** [comment]
Good idea to provide a way for getting the weather. But I think here it's not the way to deal with this case.

I think it would be better to get all info from the api (today, tomorrow and all other thing we can get with a free subscription) and return a single dict.
The user can then use the template of his choice to translate the output into his own language.

File: openweathermap.py (hunk 7 above)
Comment (corus87) [reply openweathermap.py]: Yes its still working, I just wanted to avoid to call the lib if only a forecast is called

File: openweathermap.py (hunk 8 above)
Comment (corus87) [reply openweathermap.py]: The thing is with the 5 day/3 hours API call, its for a forecast and with the pyown lib we used befor we getting the current weather and the weather of tomorrow, these are two different API´s we are calling thats why I was choosing this way

Comment (Sispheor) [reply openweathermap.py]: Yep, but all info could be placed into a single dict. 
i don't have any proposal for the implementation, but it's how all kalliope neurons work so far. So we need to find a good data representation that can work with all cases. (getting today, tomorrow, a specific day).

File: openweathermap.py [hunk 9]
```diff
@@ -1,10 +1,14 @@
+# -*- coding: utf-8 -*-
 import pyowm
 import requests
 import json
+import sys
 from datetime import datetime
 
 from kalliope.core.NeuronModule import NeuronModule, MissingParameterException
 
+reload(sys)
```
Comment (Sispheor) [inline openweathermap.py]: this should only be done if python 2 detected. Also it is done by Kalliope core before. Why did you add? You went though an issue with encoding?
